Firestore/core/src/firebase/firestore/model/no_document.cc

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/model/no_document.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+NoDocument::NoDocument(const DocumentKey& key, const SnapshotVersion& version)
+    : MaybeDocument(key, version) {
+  set_type(Type::NoDocument);
```

Comment:
Should type just be a constructor parameter?

Reply:
done